### PR TITLE
En-6080 : add unit tests consensus message validator

### PR DESCRIPTION
**Pull request by miiu96**

### CODE REVIEW

File: consensus/spos/consensusMessageValidator_test.go [hunk 1]
```diff
@@ -1,3 +1,455 @@
...
+
+	workerArgs := createDefaultWorkerArgs()
+	wrk, _ := spos.NewWorker(workerArgs)
+
+	cnsMsg := &consensus.Message{PubKeysBitmap: []byte("0")}
+	err := wrk.CheckMessageWithFinalInfoValidity(cnsMsg)
+	assert.True(t, errors.Is(err, spos.ErrInvalidPublicKeyBitmapSize))
+}
+
+func TestCheckMessageWithFinalInfoValidity_InvalidSignatureSize(t *testing.T) {
```
Comment (SebastianMarian) [inline consensus/spos/consensusMessageValidator_test.go:48]: InvalidAggregateSignatureSize

File: consensus/spos/consensusMessageValidator_test.go [hunk 2]
```diff
@@ -1,3 +1,455 @@
...
+
+	workerArgs := createDefaultWorkerArgs()
+	wrk, _ := spos.NewWorker(workerArgs)
+
+	cnsMsg := &consensus.Message{SignatureShare: []byte("0")}
+	err := wrk.CheckMessageWithSignatureValidity(cnsMsg)
+	assert.True(t, errors.Is(err, spos.ErrInvalidSignatureSize))
+}
+
+func TestCheckMessageWithSignatureValidity_ShouldWork(t *testing.T) {
```
Comment (SebastianMarian) [inline consensus/spos/consensusMessageValidator_test.go:107]: SignatureShare

File: consensus/spos/consensusMessageValidator_test.go [hunk 3]
```diff
@@ -1,3 +1,455 @@
...
+
+	workerArgs := createDefaultWorkerArgs()
+	wrk, _ := spos.NewWorker(workerArgs)
+
+	cnsMsg := &consensus.Message{SignatureShare: []byte("0")}
+	err := wrk.CheckMessageWithBlockHeaderValidity(cnsMsg)
+	assert.True(t, errors.Is(err, spos.ErrInvalidMessage))
+}
+
+func TestCheckMessageWithBlockHeaderValidity_InvalidHeaderSize(t *testing.T) {
```
Comment (SebastianMarian) [inline consensus/spos/consensusMessageValidator_test.go:131]: You can also add a test for header size larger than 1MB exactly as you did for block body size

File: consensus/spos/consensusMessageValidator_test.go [hunk 4]
```diff
@@ -1,3 +1,455 @@
...
+	assert.True(t, errors.Is(err, spos.ErrInvalidHeaderSize))
+}
+
+func TestCheckMessageWithBlockHeaderValidity_ShouldWork(t *testing.T) {
+	t.Parallel()
+
+	workerArgs := createDefaultWorkerArgs()
+	wrk, _ := spos.NewWorker(workerArgs)
+
+	cnsMsg := &consensus.Message{Header: []byte("header")}
```
Comment (SebastianMarian) [inline consensus/spos/consensusMessageValidator_test.go:148]: You can use the same pattern used below for body size:

headerBytes := make([]byte, 100)
_, _ = rand.Read(headerBytes)
cnsMsg := &consensus.Message{Header: headerBytes}

File: consensus/spos/consensusMessageValidator_test.go [hunk 5]
```diff
@@ -1,3 +1,455 @@
...
+	wrk, _ := spos.NewWorker(workerArgs)
+
+	bodyBytes := make([]byte, core.MegabyteSize+1)
+	_, _ = rand.Read(bodyBytes)
+	cnsMsg := &consensus.Message{Body: bodyBytes}
+	err := wrk.CheckMessageWithBlockBodyAndHeaderValidity(cnsMsg)
+	assert.True(t, errors.Is(err, spos.ErrInvalidBodySize))
+}
+
+func TestCheckMessageWithBlockBodyAndHeaderValidity_InvalidHeaderSize(t *testing.T) {
```
Comment (SebastianMarian) [inline consensus/spos/consensusMessageValidator_test.go:214]: You also can add a unit test for header size = 0

File: consensus/spos/consensusMessageValidator_test.go [hunk 6]
```diff
@@ -1,3 +1,455 @@
...
+
+	workerArgs := createDefaultWorkerArgs()
+	wrk, _ := spos.NewWorker(workerArgs)
+
+	cnsMsg := &consensus.Message{MsgType: int64(bls.MtUnknown), Header: []byte("1")}
+	err := wrk.CheckConsensusMessageValidityForMessageType(cnsMsg)
+	assert.True(t, errors.Is(err, spos.ErrInvalidMessageType))
+}
+
+func TestSsBlockHeaderHashSizeValid_NotValid(t *testing.T) {
```
Comment (SebastianMarian) [inline consensus/spos/consensusMessageValidator_test.go:306]: TestIs

File: consensus/spos/consensusMessageValidator_test.go [hunk 7]
```diff
@@ -1,3 +1,455 @@
...
+
+	workerArgs := createDefaultWorkerArgs()
+	wrk, _ := spos.NewWorker(workerArgs)
+
+	cnsMsg := &consensus.Message{MsgType: int64(bls.MtBlockBody), BlockHeaderHash: []byte("hash")}
+	result := wrk.IsBlockHeaderHashSizeValid(cnsMsg)
+	assert.False(t, result)
+}
+
+func TestSsBlockHeaderHashSizeValid(t *testing.T) {
```
Comment (SebastianMarian) [inline consensus/spos/consensusMessageValidator_test.go:317]: TestIs

File: consensus/spos/consensusMessageValidator_test.go [hunk 8]
```diff
@@ -1,3 +1,455 @@
...
+
+	cnsMsg := &consensus.Message{
+		ChainID: chainID, MsgType: int64(bls.MtBlockBodyAndHeader),
+		Header: headerBytes, BlockHeaderHash: headerHash, PubKey: pubKey, Signature: signature, RoundIndex: 10,
+	}
+	err := wrk.CheckConsensusMessageValidity(cnsMsg)
+	assert.True(t, errors.Is(err, spos.ErrMessageForFutureRound))
+}
+
+func TestCheckConsensusMessageValidity_ErrMessageForPastRound(t *testing.T) {
```
Comment (SebastianMarian) [inline consensus/spos/consensusMessageValidator_test.go:434]: You can add two more tests: TestCheckConsensusMessageValidity_InvalidSignature and TestCheckConsensusMessageValidity_Ok

File: consensus/spos/consensusMessageValidator_test.go (hunk 1 above)
Comment (miiu96) [reply consensus/spos/consensusMessageValidator_test.go:48]: done.

File: consensus/spos/consensusMessageValidator_test.go (hunk 2 above)
Comment (miiu96) [reply consensus/spos/consensusMessageValidator_test.go:107]: done.

File: consensus/spos/consensusMessageValidator_test.go (hunk 3 above)
Comment (miiu96) [reply consensus/spos/consensusMessageValidator_test.go:131]: done.

File: consensus/spos/consensusMessageValidator_test.go (hunk 4 above)
Comment (miiu96) [reply consensus/spos/consensusMessageValidator_test.go:148]: done.

File: consensus/spos/consensusMessageValidator_test.go (hunk 5 above)
Comment (miiu96) [reply consensus/spos/consensusMessageValidator_test.go:214]: done.

File: consensus/spos/consensusMessageValidator_test.go (hunk 6 above)
Comment (miiu96) [reply consensus/spos/consensusMessageValidator_test.go:306]: done.

File: consensus/spos/consensusMessageValidator_test.go (hunk 7 above)
Comment (miiu96) [reply consensus/spos/consensusMessageValidator_test.go:317]: done.

File: consensus/spos/consensusMessageValidator_test.go (hunk 8 above)
Comment (miiu96) [reply consensus/spos/consensusMessageValidator_test.go:434]: done.

File: consensus/spos/consensusMessageValidator_test.go [hunk 9]
```diff
@@ -104,7 +106,7 @@ func TestCheckMessageWithSignatureValidity_InvalidSignatureShareSize(t *testing.
 	assert.True(t, errors.Is(err, spos.ErrInvalidSignatureSize))
 }
 
-func TestCheckMessageWithSignatureValidity_ShouldWork(t *testing.T) {
+func TestCheckMessageWithSignatureValidity_SignatureShareShouldWork(t *testing.T) {
```
Comment (SebastianMarian) [inline consensus/spos/consensusMessageValidator_test.go:109]: I think that TestCheckMessageWithSignatureShareValidity_ShouldWork, is ok

File: consensus/spos/consensusMessageValidator_test.go [hunk 10]
```diff
@@ -139,6 +141,31 @@ func TestCheckMessageWithBlockHeaderValidity_InvalidHeaderSize(t *testing.T) {
 	assert.True(t, errors.Is(err, spos.ErrInvalidHeaderSize))
 }
 
+func TestCheckMessageWithBlockHeaderValidity_HeaderToBig(t *testing.T) {
```
Comment (SebastianMarian) [inline consensus/spos/consensusMessageValidator_test.go:144]: TooBig

File: consensus/spos/consensusMessageValidator_test.go (hunk 9 above)
Comment (miiu96) [reply consensus/spos/consensusMessageValidator_test.go:109]: done.

File: consensus/spos/consensusMessageValidator_test.go (hunk 10 above)
Comment (miiu96) [reply consensus/spos/consensusMessageValidator_test.go:144]: done.

File: consensus/spos/consensusMessageValidator_test.go [hunk 11]
```diff
@@ -1,3 +1,537 @@
...
+	workerArgs := createDefaultWorkerArgs()
+	wrk, _ := spos.NewWorker(workerArgs)
+
+	headerBytes := make([]byte, 100)
+	_, _ = rand.Read(headerBytes)
+	headerHash := make([]byte, workerArgs.Hasher.Size())
+	_, _ = rand.Read(headerHash)
+	pubKey := make([]byte, PublicKeySize)
+	_, _ = rand.Read(pubKey)
+	signature := make([]byte, SignatureSize)
```
Comment (bogdan-rosianu) [inline consensus/spos/consensusMessageValidator_test.go:430]: shadowing variable

File: consensus/spos/consensusMessageValidator_test.go [hunk 12]
```diff
@@ -1,3 +1,537 @@
...
+
+	workerArgs := createDefaultWorkerArgs()
+	wrk, _ := spos.NewWorker(workerArgs)
+
+	headerBytes := make([]byte, 100)
+	_, _ = rand.Read(headerBytes)
+	headerHash := make([]byte, workerArgs.Hasher.Size())
+	_, _ = rand.Read(headerHash)
+	pubKey := []byte(workerArgs.ConsensusState.ConsensusGroup()[0])
+	signature := make([]byte, SignatureSize)
```
Comment (bogdan-rosianu) [inline consensus/spos/consensusMessageValidator_test.go:452]: shadowing variable

File: consensus/spos/consensusMessageValidator_test.go [hunk 13]
```diff
@@ -1,3 +1,537 @@
...
+	workerArgs := createDefaultWorkerArgs()
+	workerArgs.ConsensusState.RoundIndex = 100
+	wrk, _ := spos.NewWorker(workerArgs)
+
+	headerBytes := make([]byte, 100)
+	_, _ = rand.Read(headerBytes)
+	headerHash := make([]byte, workerArgs.Hasher.Size())
+	_, _ = rand.Read(headerHash)
+	pubKey := []byte(workerArgs.ConsensusState.ConsensusGroup()[0])
+	signature := make([]byte, SignatureSize)
```
Comment (bogdan-rosianu) [inline consensus/spos/consensusMessageValidator_test.go:475]: shadowing variable

File: consensus/spos/consensusMessageValidator_test.go [hunk 14]
```diff
@@ -1,3 +1,537 @@
...
+	workerArgs.SingleSigner = signer
+	workerArgs.ConsensusState.RoundIndex = 10
+	wrk, _ := spos.NewWorker(workerArgs)
+
+	headerBytes := make([]byte, 100)
+	_, _ = rand.Read(headerBytes)
+	headerHash := make([]byte, workerArgs.Hasher.Size())
+	_, _ = rand.Read(headerHash)
+	pubKey := []byte(workerArgs.ConsensusState.ConsensusGroup()[0])
+	signature := make([]byte, SignatureSize)
```
Comment (bogdan-rosianu) [inline consensus/spos/consensusMessageValidator_test.go:505]: shadowing variable

File: consensus/spos/consensusMessageValidator_test.go [hunk 15]
```diff
@@ -1,3 +1,537 @@
...
+	workerArgs := createDefaultWorkerArgs()
+	workerArgs.ConsensusState.RoundIndex = 10
+	wrk, _ := spos.NewWorker(workerArgs)
+
+	headerBytes := make([]byte, 100)
+	_, _ = rand.Read(headerBytes)
+	headerHash := make([]byte, workerArgs.Hasher.Size())
+	_, _ = rand.Read(headerHash)
+	pubKey := []byte(workerArgs.ConsensusState.ConsensusGroup()[0])
+	signature := make([]byte, SignatureSize)
```
Comment (bogdan-rosianu) [inline consensus/spos/consensusMessageValidator_test.go:528]: shadowing variable

File: consensus/spos/consensusMessageValidator_test.go (hunk 11 above)
Comment (miiu96) [reply consensus/spos/consensusMessageValidator_test.go]: done.

File: consensus/spos/consensusMessageValidator_test.go (hunk 12 above)
Comment (miiu96) [reply consensus/spos/consensusMessageValidator_test.go]: done.

File: consensus/spos/consensusMessageValidator_test.go (hunk 13 above)
Comment (miiu96) [reply consensus/spos/consensusMessageValidator_test.go]: done.

File: consensus/spos/consensusMessageValidator_test.go (hunk 14 above)
Comment (miiu96) [reply consensus/spos/consensusMessageValidator_test.go]: done.

File: consensus/spos/consensusMessageValidator_test.go (hunk 15 above)
Comment (miiu96) [reply consensus/spos/consensusMessageValidator_test.go]: done.

**Pull request merged**